test/spec/directives/datepicker.spec.js

```diff
@@ -18,19 +18,33 @@ describe('Directive: datepicker', function () {
     scope = element.isolateScope() || element.scope();
   }));
 
-  it('Sets the date to the correct format based on the type of date question', function () {
+  it('Sets the date to the correct format for the graduation date question datepicker', function () {
+    scope.monthYearOnly = true;
     scope.updateTimeStamp(new Date('02/05/1994'));
 
     expect(scope.localModel).toBe('1994-02-10');
 
+    scope.updateTimeStamp(new Date('02/05/1994 10:11:12'));
+
+    //removes time and sets the day to 10
+    expect(scope.localModel).toBe('1994-02-10');
+
+    expect(scope.dateOptions.viewMode).toBe('years');
+  });
+
+  it('Sets the date to the correct format for all other types of datepickers', function () {
     scope.monthYearOnly = false;
 
     scope.updateTimeStamp(new Date('02/05/1994'));
 
+    //adds the time and reformats
     expect(scope.localModel).toBe('1994-02-05 00:00:00');
   });
 
-  it('Sets date options correctly based on type of date', function () {
-    expect(scope.dateOptions.viewMode).toBe('years');
+  it('Sets the initial/default date to the correct value and format based on the localModel', function () {
+    let localModel = new Date('01-30-2023 11:01:05');
+    let dateOptions = scope.getDateOptions(localModel);
+
+    expect(dateOptions.defaultDate).toBe('2023-01-30 11:01:05');
```

Comment:
Sorry to be a stickler about this, but can you add a few more tests to ensure `getDateOptions` is acting correctly?

- Ensure the whole object comes back the way you intend.
- If the API returns month & year dates differently, a test with a date formatted like that.
- Test without month & year to ensure getDateOptions returns correctly.

Reply:
done